astropy/time/core.py

```diff
@@ -282,6 +283,8 @@ def _get_time_fmt(self, val, val2, format, scale):
             try:
                 return FormatClass(val, val2, scale, self.precision,
                                    self.in_subfmt, self.out_subfmt)
+            except UnitConversionError:
```

Comment:
Ah, let me clarify, the reason this needs to be caught and handled separately is that there were some tests where a `UnitsError` was expected to be raised.  Now a `UnitConversionError` is raised, but since that's a subclass of `ValueError` it needs to be handled separately.  

I can't remember exactly what the use case was but there is code expecting this to raise unit conversion errors so that they can be handled separately.